components/script/dom/imagedata.rs

```diff
@@ -64,8 +64,9 @@ impl ImageData {
     ) -> Fallible<DomRoot<ImageData>> {
         // checking jsobject type
         let cx = GlobalScope::get_cx();
-        typedarray!(in(*cx) let array_res: Uint8ClampedArray = jsobject);
-        let array = array_res.map_err(|_| {
+        let heap_typed_array =
+            new_initialized_heap_typed_array(HeapTypedArrayInit::Object(jsobject));
+        let array = heap_typed_array.acquire_data(cx).map_err(|_| {
```

Comment:
`acquire_data` detaches the buffer, so instead we want to use `get_internal`.

Reply:
Indeed thanks, fixed